tests/store/artifact/test_azure_data_lake_artifact_repo.py

```diff
@@ -0,0 +1,229 @@
+TEST_DATA_LAKE_URI_BASE = "abfss://filesystem@account.dfs.core.windows.net"
+TEST_DATA_LAKE_URI = posixpath.join(TEST_DATA_LAKE_URI_BASE, "some/path")
+
+
+class MockPathList:
+    def __init__(self, items, next_marker=None):
+        self.items = items
+        self.next_marker = next_marker
+
+    def __iter__(self):
+        return iter(self.items)
+
+
+@pytest.fixture
+def mock_data_lake_client():
+    mock_adls_client = mock.MagicMock(autospec=DataLakeServiceClient)
+    with mock.patch(
+        "mlflow.store.artifact.azure_data_lake_artifact_repo._get_data_lake_client"
+    ) as mock_get_client:
+        mock_get_client.return_value = mock_adls_client
+        yield mock_adls_client
+
+
+@pytest.fixture
+def mock_filesystem_client(mock_data_lake_client):
+    mock_fs_client = mock.MagicMock(autospec=FileSystemClient)
+    mock_data_lake_client.get_file_system_client.return_value = mock_fs_client
+    yield mock_fs_client
+
+
+@pytest.fixture
+def mock_directory_client(mock_filesystem_client):
+    mock_directory_client = mock.MagicMock(autospec=DataLakeDirectoryClient)
+    mock_filesystem_client.get_directory_client.return_value = mock_directory_client
+    yield mock_directory_client
+
+
+@pytest.fixture
+def mock_file_client(mock_directory_client):
+    mock_file_client = mock.MagicMock(autospec=DataLakeFileClient)
+    mock_directory_client.get_file_client.return_value = mock_file_client
+    yield mock_file_client
+
+
+def test_parse_global_abfss_uri():
+    parse = _parse_abfss_uri
+    global_abfs_with_short_path = "abfss://filesystem@acct.dfs.core.windows.net/path"
+    assert parse(global_abfs_with_short_path) == ("filesystem", "acct", "path")
+
+    global_abfs_without_path = "abfss://filesystem@acct.dfs.core.windows.net"
+    assert parse(global_abfs_without_path) == ("filesystem", "acct", "")
+
+    global_abfs_without_path2 = "abfss://filesystem@acct.dfs.core.windows.net/"
+    assert parse(global_abfs_without_path2) == ("filesystem", "acct", "")
+
+    global_abfs_with_multi_path = "abfss://filesystem@acct.dfs.core.windows.net/a/b"
+    assert parse(global_abfs_with_multi_path) == ("filesystem", "acct", "a/b")
+
+    with pytest.raises(MlflowException, match="ABFSS URI must be of the form"):
+        parse("abfss://filesystem@acct.dfs.core.evil.net/path")
+    with pytest.raises(MlflowException, match="ABFSS URI must be of the form"):
+        parse("abfss://filesystem@acct/path")
+    with pytest.raises(MlflowException, match="ABFSS URI must be of the form"):
+        parse("abfss://acct.dfs.core.windows.net/path")
+    with pytest.raises(MlflowException, match="ABFSS URI must be of the form"):
+        parse("abfss://@acct.dfs.core.windows.net/path")
+    with pytest.raises(MlflowException, match="ABFSS URI must be of the form"):
+        parse("abfss://filesystem@acctxdfs.core.windows.net/path")
+    with pytest.raises(MlflowException, match="Not an ABFSS URI"):
+        parse("abfs://cont@acct.dfs.core.windows.net/path")
```

Comment:
```suggestion
@pytest.mark.paramerize(...)
def test_parse_global_abfss_uri(uri):
    with pytest.raises(MlflowException, match="ABFSS URI must be of the form"):
        parse(uri)
```

Can we parametrize this test?